src/main/java/frc/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,362 @@
+    double armDemandVoltsB =
+        ArmCal.ARM_FEEDFORWARD.calculate(armController.getSetpoint().velocity);
+    double armDemandVoltsC = ArmCal.ARBITRARY_ARM_FEED_FORWARD_VOLTS * getCosineArmAngle();
+    armMotor.setVoltage(armDemandVoltsA + armDemandVoltsB + armDemandVoltsC);
+
+    SmartDashboard.putNumber("Arm PID", armDemandVoltsA);
+    SmartDashboard.putNumber("Arm FF", armDemandVoltsB);
+    SmartDashboard.putNumber("Arm Gravity", armDemandVoltsC);
+  }
+
+  /**
+   * if the robot has completed startScore() but hasn't started finishScore, then stop the robot from
+   * scoring
+   */
+  public void cancelScore() {
+    if (scoringInProgress) {
+      setCancelScore(true);
+    }
+  }
+
+  /** Runs instead of finishScore if cancelScore is true. */
+  public void finishScoreCancelled() {
+    setCancelScore(false);
+    ManualPrepScoreSequence();
+  }
+
+  /** returns cancelScore (true if scoring action is cancelled) */
+  public boolean getCancelScore() {
+    return this.cancelScore;
+  }
+
+  /** sets cancelScore (true if scoring action is cancelled) */
+  public void setCancelScore(boolean cancelled) {
+    this.cancelScore = cancelled;
+  }
+
+  /** sets scoringInProgress */
+  public void setScoringInProgress(boolean isScoring) {
+    scoringInProgress = isScoring;
+  }
+
+  /**
+   * takes the column and height from ScoringLocationUtil.java and converts that to a ArmPosition
+   * then gives the position to the given arm
+   */
+  public void ManualPrepScoreSequence() {
+    ScoreHeight height = scoreLoc.getScoreHeight();
+
+    // low for all columns is the same height
+    if (height == ScoreHeight.LOW) {
+      setDesiredPosition(ArmPosition.SCORE_LOW);
+    } else {
+      setDesiredPosition(ArmPosition.SCORE_MID_HIGH);
+    }
+  }
+
+  public void zeroArmAtCurrentPos() {
+    ArmCal.ARM_ABSOLUTE_ENCODER_ZERO_POS_DEG = armAbsoluteEncoder.getPosition();
+    System.out.println("New Zero for Arm: " + ArmCal.ARM_ABSOLUTE_ENCODER_ZERO_POS_DEG);
+  }
+
+  private REVLibError setDegreesFromGearRatio(
+        AbsoluteEncoder sparkMaxEncoder, double ratio) {
+      double degreesPerRotation = 360.0 / ratio;
+      double degreesPerRotationPerSecond = degreesPerRotation / 60.0;
+      REVLibError error = sparkMaxEncoder.setPositionConversionFactor(degreesPerRotation);
+
+      if (error != REVLibError.kOk) {
+        return error;
+      }
+
+      return sparkMaxEncoder.setVelocityConversionFactor(degreesPerRotationPerSecond);
+    }
+
+    public static REVLibError setDegreesFromGearRatio(
+        RelativeEncoder sparkMaxEncoder, double ratio) {
+      double degreesPerRotation = 360.0 / ratio;
+      double degreesPerRotationPerSecond = degreesPerRotation / 60.0;
+      REVLibError error = sparkMaxEncoder.setPositionConversionFactor(degreesPerRotation);
+
+      if (error != REVLibError.kOk) {
+        return error;
+      }
+
+      return sparkMaxEncoder.setVelocityConversionFactor(degreesPerRotationPerSecond);
+    }
```

Comment:
Why do you hace duplicates (except one is `private` and one is `public static`)?

Reply:
One is for Relative Encoders and one is for Absolute Encoders. I renamed the methods to be more clear.